exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'uri'
+
+module OpenTelemetry
+  module Exporter
+    module Jaeger
+      # An OpenTelemetry trace exporter that sends spans over HTTP as Thrift Binary encoded Jaeger spans.
+      class CollectorExporter
+        SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+        FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+        private_constant(:SUCCESS, :FAILURE)
+
+        def initialize(endpoint: ENV.fetch('OTEL_EXPORTER_JAEGER_ENDPOINT', 'http://localhost:14250'),
+                       username: ENV['OTEL_EXPORTER_JAEGER_USER'],
+                       password: ENV['OTEL_EXPORTER_JAEGER_PASSWORD'])
+          raise ArgumentError, "invalid url for Jaeger::CollectorExporter #{endpoint}" if invalid_url?(endpoint)
+
+          transport = ::Thrift::HTTPClientTransport.new(endpoint) # TODO: how to specify username and password?
```

Comment:
Looks like you'll want to use the `add_headers` method on the transport client https://github.com/apache/thrift/blob/master/lib/rb/lib/thrift/transport/http_client_transport.rb#L41-L43

You will be using the `Authorization` header, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

I have not setup basic auth in a while someone else feel free to jump in and correct me but I think the code will look something like this.

```rb
transport = ::Thrift::HTTPClientTransport.new(endpoint)
authorization = Base64.strict_encode64("#{username}:#{password}")
auth_header = { 'Authorization': "Basic #{authorization}" }
transport.add_headers(auth_header)
```